src/lib/parts.ts

```diff
@@ -24,9 +24,21 @@ import {TemplateInstance} from './template-instance.js';
 import {TemplateResult} from './template-result.js';
 import {createMarker} from './template.js';
 
-export const isPrimitive = (value: any) =>
-    (value === null ||
-     !(typeof value === 'object' || typeof value === 'function'));
+// https://tc39.github.io/ecma262/#sec-typeof-operator
+export type Primitive = null|undefined|boolean|number|string|Symbol|bigint;
+export const isPrimitive = (value: unknown): value is Primitive => {
+  return (
+      value === null ||
+      !(typeof value === 'object' || typeof value === 'function'));
+};
+
+const isIterable = (value: unknown): value is Iterable<unknown> => {
+  return !!(
```

Comment:
why the `!!`?

Reply:
To ensure that it's a boolean. Reverted in any case.